src/main/java/org/folio/service/inventory/InventoryManager.java

```diff
@@ -1265,4 +1277,24 @@ private void updateItemWithPieceFields(Piece piece, JsonObject item) {
     Optional.ofNullable(piece.getDiscoverySuppress())
       .ifPresentOrElse(discSup -> item.put(ITEM_DISCOVERY_SUPPRESS, discSup), () -> item.remove(ITEM_DISCOVERY_SUPPRESS));
   }
+
+  public Future<SharingInstance> createShadowInstanceIfNeeded(String instanceId, RequestContext requestContext) {
+    return consortiumConfigurationService.getConsortiumConfiguration(requestContext)
+      .compose(consortiumConfiguration -> {
+        if (consortiumConfiguration.isPresent()) {
+          return getInstanceById(instanceId, true, requestContext)
+            .compose(instance -> {
+              if (Objects.nonNull(instance) && !instance.isEmpty()) {
+                return Future.succeededFuture();
+              }
+              logger.info("Creating shadow instance with instanceId: {}", instanceId);
+              return sharingInstanceService.createShadowInstance(instanceId, consortiumConfiguration.get(), requestContext);
+            });
+        }
+        logger.warn("ECS mode is not enabled for tenant with id {} when creating shadow instance with id {}",
+          TenantTool.tenantId(requestContext.getHeaders()), instanceId);
```

Comment:
I think we don't need this warn log, if we are not in ECS mode - so consortiumConfiguration will be never presented and it is normal situation

Reply:
removed